### PR TITLE
Feature/dev 1830 detect native coreos

### PR DESCRIPTION
**Pull request by JoeReid**

### CODE REVIEW

File: cmd/storageos/storageos.go [hunk 1]
```diff
@@ -35,6 +37,38 @@ func init() {
...
+}
+
+func verfyHostPlatform() error {
+	// Detect native execution on coreOS
+	// coreOS should not run user-land programs, and we will not work there (outside a container)
+	if coreos, err := isCoreOS(); err == nil && coreos {
+
+		// If we dont think we are in a container, fail and warn the user
+		if inContainer, err := isInContainer(); err == nil && !inContainer {
+			return errors.New("do not run user-land utilities directly on coreOS, please use the CLI container")
```
Comment (oicheryl) [inline cmd/storageos/storageos.go]: To use the StorageOS CLI on Container Linux, run:

docker run --rm -e STORAGEOS_HOST -e STORAGEOS_USERNAME -e STORAGEOS_PASSWORD storageos/cli

Comment (JoeReid) [reply cmd/storageos/storageos.go]: @oicheryl After a chat with Simon, it seems that we are not really set-up for supporting multi-line errors. I agree that your message is far better, however it is outside the scope of this PR to add proper support for multi-line messages.

I am happy to take a ticket for this feature should one exist (be created).

Comment (oicheryl) [reply cmd/storageos/storageos.go]: Then I would replace with "To use the StorageOS CLI on Container Linux, you need to run the storageos/cli image."

Comment (JoeReid) [reply cmd/storageos/storageos.go]: I have made this change, I will treat this PR as merge-able.

**Pull request merged**